### PR TITLE
Bulk update UI

### PR DESCRIPTION
**Pull request by Mvalverde00**
Client side support for bulk updating.  When the user clicks on a bottom-level pie chart slice or a table row, a modal pops up prompting the user to enter information to perform a bulk update.  The user is told how many devices they are updating and of what kind, and warned that once started this action is irreversible.

### CODE REVIEW

**Review by georgezheng999 — APPROVED** [review]

File: src/main/webapp/dashboard-manager.js [hunk 1]
```diff
@@ -15,6 +15,16 @@ class DashboardManager {
     google.visualization.events.addListener(
         this.aggregationSelector, 'statechange', this.updateAndDrawData.bind(this));
 
+    document.addEventListener('bulkUpdate', (e) => {
+      let row = e.detail;
+      let deviceIds = this.data.getValue(row, this.data.getNumberOfColumns() - 2);
+      let selectedValues = [...Array(this.aggregationSelector.getState().selectedValues.length).keys()].map(index => this.data.getValue(row, index));
```
Comment (qlyao) [inline src/main/webapp/dashboard-manager.js]: consider making it multiple lines

File: src/main/webapp/dashboard-manager.js [hunk 2]
```diff
@@ -15,6 +15,16 @@ class DashboardManager {
     google.visualization.events.addListener(
         this.aggregationSelector, 'statechange', this.updateAndDrawData.bind(this));
 
+    document.addEventListener('bulkUpdate', (e) => {
+      let row = e.detail;
+      let deviceIds = this.data.getValue(row, this.data.getNumberOfColumns() - 2);
```
Comment (qlyao) [inline src/main/webapp/dashboard-manager.js]: Can `this.data.getNumberOfColumns() - 2` be defined as a constant? It's hard for others who are less familiar with the code to understand why there is a `2` here.

Here and below.

File: src/main/webapp/dashboard-manager.js [hunk 3]
```diff
@@ -68,11 +80,19 @@ class DashboardManager {
...
         .then(response => {
             let results = response.response;
-            for (let entry of results) {
+            for (let [index, entry] of results.entries()) {
               const row =
                   selectorState.map(displayName => entry[getAnnotatedFieldFromDisplay(displayName).API]);
               row.push(entry.count);
 
+              if (entry.deviceIds == null) {
+                entry.deviceIds = ['abc', 'def', 'ghi', 'jkl', 'mno'];
```
Comment (qlyao) [inline src/main/webapp/dashboard-manager.js:96]: Is it for local testing? If so, can you add a TODO to remove it?

File: src/main/webapp/dashboard-manager.js [hunk 4]
```diff
@@ -102,7 +125,8 @@ class DashboardManager {
     let result = google.visualization.data.group(
         filtered,
         [depth - 1],
-        [{'column': filtered.getNumberOfColumns() - 1, 'aggregation': google.visualization.data.sum, 'type': 'number'}]);
+                    // Devices count is second to last column
```
Comment (qlyao) [inline src/main/webapp/dashboard-manager.js]: The indent of the comment looks weird. If you decide to make `getNumberOfColumns() - 2` a constant as mentioned in a previous comment, then this comment can be removed.

**Review by qlyao — APPROVED** [review]

File: src/main/webapp/table-manager.js [hunk 5]
```diff
@@ -86,6 +88,20 @@ class TableManager {
     }
   }
 
+  setTableView(numOfCols) {
+    /* Aggregation table has Field 1 | ... | Field n | deviceCount | deviceIds | Button
```
Comment (ChuxiangP) [inline src/main/webapp/table-manager.js]: nit: leave one line above?

```
/*
 * Aggregation table ...
 ...
```

**Review by ChuxiangP — APPROVED** [review]
Oops, sorry, missed this one.

**Pull request merged**